buildSrc/src/main/java/edu/wpi/gripgenerator/settings/DefinedParamType.java

```diff
@@ -41,7 +41,7 @@ public boolean isOutput() {
     }
 
     private final String type;
-    private DefinedParamState state;
+    private Optional<DefinedParamState> state;
```

Comment:
Why is this an Optional?  Shouldn't the state be required?  And why is it called "state" and not "direction"?